x-pack/plugins/synthetics/common/formatters/formatting_utils.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { replaceVarsWithParams, ParsedVars } from './lightweight_param_formatter';
+import variableParser from './variable_parser';
+import { ConfigKey, MonitorFields } from '../runtime_types';
+
+export type FormatterFn = (fields: Partial<MonitorFields>, key: ConfigKey) => string | null;
+
+export const arrayToJsonFormatter: FormatterFn = (fields, key) => {
+  const value = (fields[key] as string[]) ?? [];
+  return value.length ? JSON.stringify(value) : null;
+};
+
+export const objectToJsonFormatter: FormatterFn = (fields, key) => {
```

Comment:
I think this one should call paramReplaceFormatter by default for sure. It's used for things like request headers.